docs/how_to/add_op_in_backend.md

```diff
@@ -0,0 +1,609 @@
+`data` comes from the return value of the user defined `FListInputName` function.
+Users still can generate an executor for the `quand_func` like the following:
+```python
+quand_exe = quand_func.simple_bind(ctx=mx.cpu(), quandratic0_data=(1,))
+```
+- Line 12: Register shape inference function.
+- Line 13: Register type inference function.
+- Line 14: Register forward function.
+- Line 15: Register the function for creating the node of the operator in
+a backward pass. Note that we used a convenience functor struct `ElemwiseGradUseIn`.
+As you can tell from the name, the registered functor creates the node for gradient computation
+with dependencies on the output gradient node and input node. Similarly, there are
+other three functors defined as `ElemwiseGradUseOut`, `ElemwiseGradUseInOut`,
+and `ElemwiseGradUseNone` for developers' convenience. In order to add
+this attribute, we also need to register a backward operator for `quadratic` with
+several basic attributes, as it can share attribute inference
+functions with the forward operator and is not exposed to frontend.
+- Lines 16-19: This registered function implies that which output tensor can reuse
+which input tensor's memory space instead of allocating a new memory space for the output.
+In the operator `quadratic`, there is only one input and output, and the output can reuse
+the input memory space, so we store a pair of zeros in the function return vector
+indicating that `inputs[0]`'s memory space can be reused by `outputs[0]`.
+Note that this function just provides a hint to the computational graph initializer.
+If there are other nodes depending on the input tensor, the memory space
+of the input tensor will not be overwritten by the output.
+- Line 20: Define the input argument name as `data` for the operator.
+- Line 21: Add user input parameters `a`, `b`, and `c` as the attributes of the operator.
+- Line 22: Register an operator named `_backward_quadratic` for backward pass
+of the operator `quadratic`. The underscore prefix in the operator name indicates
+that this is an operator not exposed to users. The convention
+of naming an internally used backward operator is prepending the prefix `_backward_`
+to the corresponding forward operator name.
+- Line 23: Set the parameter parser for the operator `_backward_quadratic`.
+- Line 24: Set the number of inputs.
+- Line 25: Set the number of outputs.
+- Line 26: Add `TIsBackward` attribute for the operator. The shape and type
+inference passes use this attribute to determine whether a node in the graph is a
+forward or backward node.
+- Line 27: Register backward function.
+
+So far, we have acquired an operator working on CPU in frontend.
+In order to register the operator working on GPUs, we just need to add the following
+code to `quadratic_op.cu`. Note that forward and backward functions
+are registered with attribute key `FCompute<gpu>`, rather than `FCompute<cpu>`.
```

Comment:
What's the difference between FCompute and FComputeEx? Where can I find any document about that?

Reply:
FComputeEx taked NDArray instead of TBlob items and is generally called when the tensor is of sparse storage type (kCSRStorage or kRowSparseStorage instead of kDefaultStorage)

Reply:
Thanks for your kindly reply, @cjolivier01. One more question, I notice that there are ForwardResource and BackwardResource methods to register addtional memory resource for computation in previous document for creating OP [here](https://mxnet.incubator.apache.org/architecture/overview.html#operator-property).
So with NNVM interfaces, how can I register these resouces for forward and backward computation? And how can I share some states/memory between forward and backward computation. It seems the two functions are defined seperately in this tutorail.

Reply:
1. For registering temp resources using NNVM interface, you can take a look at this example. You need to register resources separately for forward and backward if they both need temp resources.
https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/matrix_op.cc#L519
2. For sharing states between forward and backward, you can take a look at BatchNorm op which has a `aux_states` argument for both forward and backward function. Please note that this kind of interface is going to be deprecated. The new way for sharing states would use NNVM interface to register shared states.

Reply:
Many thanks, @reminisce. I will look into the aux_states in BN and the new NNVM interfaces. So what's your suggestion about stateless ops .vs. stateful ops for MXNet?

Reply:
I suggest focusing on nnvm interface for both stateful and stateless ops from now on as the legacy interface is going to become deprecated. You can follow this PR for more details on using nnvm interface for stateful ops.
https://github.com/apache/incubator-mxnet/pull/8302

Reply:
I see. Thanks for your help. :)